subprojects/build-cache/src/main/java/org/gradle/caching/internal/controller/service/DefaultBuildCacheEntryInternal.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.caching.internal.controller.service;
+
+import org.gradle.caching.internal.BuildCacheEntryInternal;
+
+import java.io.File;
+
+public class DefaultBuildCacheEntryInternal implements BuildCacheEntryInternal {
+
+    private final File file;
+
+    public DefaultBuildCacheEntryInternal(File file) {
+        this.file = file;
+    }
+
+    @Override
+    public File getFile() {
+        return file;
+    }
+
+    @Override
+    public void markDownloading() {
```

Comment:
Having this here is quite weird. I'm trying to come up with something better, but haven't had any success.